### ISSUE TITLE
About "External" of "Link from"

### ISSUE THREAD

**Issue by HIROORIH** [danmunn/redmine_dmsf#956]
Hello.

there is a question about the URL of "External" of "Link from".

I am trying to register a URL containing Japanese.
However, if you include Japanese, it will result in an error.
The solution was to convert Japanese to UTF - 8.

The number of characters that can be set in the URL is displayed as 255 characters (length) (Japan environment)

However, converting Japanese to utf8 will greatly increase the number of characters.
With 255 letters most links can not be created.

I want to cancel this limitation of 255 characters so that even 1000 characters, for example, can be registered.

I am watching the DMSF_link controller etc, but I do not know where the restriction is applied.

Is not it restricted by the DMSF plugin?
Perhaps, Redmine itself? Or is it Ruby on Rails · · ·.

For Mysql's varchar (255) and varchar (256) and above, we are aware of the differences that require 1 byte and 2 bytes, but I'd like to be able to register for the time being.

Please lend me your help.

I am sorry if It had the same question in the past.
I searched but I could not find it.

**Comment by HIROORIH** [comment]
I will add.

I realized that the place to write is different.
I overlooked that the model had a limit of 255.

After changing the DB and changing the model, I registered.

【procedure】
(1) The definition of the "external_url" item in the "dmsf_links" table of the DB is changed to varchar (255) to varchar (2550).
② "validates_length_of: external_url,: maximum => 255" of "dmsf_link.rb" has been changed to 2550.

You can now register, but when you return to the DMSF screen it has become the image state.

![redmine_dmsf_longurl1](https://user-images.githubusercontent.com/44634822/51584873-899eb080-1f1a-11e9-8d94-03d7f4f36218.PNG)

dmsf_filename has been lengthened and disappeared.
When you shrink the screen, the usual menu appeared on the right side.

For the time being, I will report it by solving dmsf_filename by hiding it.
③ "$ (". Dmsf_filename "). Hide ();" has been added to javascript.
ERROR etc are not happening so far.




I could do what I wanted to do above.
If there is a problem with the above operation, I would like advice, but is it possible?

**Comment by picman** [comment]
Can you post an example of URL which makes problems, please?

**Comment by HIROORIH** [comment]
example of URL:
https://www.google.com/search?q=寿司
(This is similar, for example if you want to register an address to search SUSHI on Google)

Just to be sure, I show it as an image.
【↓URLimage↓】
![image](https://user-images.githubusercontent.com/44634822/51598145-32abd200-1f40-11e9-93a8-906e723c996a.png)


As in the image below, Japanese enters the form as it is.
![image](https://user-images.githubusercontent.com/44634822/51597405-83bac680-1f3e-11e9-98f2-e69ca8282ee1.png)
When registering here,
It is pointed out that the URL has a problem.
![image](https://user-images.githubusercontent.com/44634822/51597551-d6947e00-1f3e-11e9-9cc3-1938a3380adf.png)

If you convert the Japanese part of the problematic URL to UTF - 8, you can register it.
URL that can be registered:
https://www.google.com/search?q=%E5%AF%BF%E5%8F%B8

However, if the converted URL exceeds 255 characters, it becomes an error and it can not be registered.
So I tried raising the upper limit this time.

Is this OK with this content?
Please tell me if there is not enough information.

**Comment by picman** [comment]
Look at the patch above. I've simply skipped the validation.

**Comment by HIROORIH** [comment]
Thank you for preparing for an early patch.
I will try it on the verification machine immediately.

By the way. .
I'm sorry but I'd like you to check if the adaptation procedure is right.
Is it OK if you mix two files "dmsf_url_validator.rb" and "dmsf_link_test.rb" that you refurbished into the source of 1.6.2 and do re-migration?

**Comment by HIROORIH** [comment]
Mr. picman.

Since there was progress, I will add it.
I tried the source I gave at 1.6.2.

A Japanese URL was hit like an image. Thank you very much.

![image](https://user-images.githubusercontent.com/44634822/51656778-c08dc880-1fe5-11e9-8226-1c56dcab23dc.png)

There is a thing to tell it for just in case.
In addition to the DMSF of 1.6.2 this time, since I am using windows server, an event of # 936 occurred.
So, I carried out the experiment with the subsequent # 938 added to the source.

However, there are things that I care about this time.
If you enter a URL of 255 length, it will look like the image below, even if the English single URL or the Japanese included URL.
IE and Chrome.
![image](https://user-images.githubusercontent.com/44634822/51656878-f8950b80-1fe5-11e9-8280-24c1b4a085d9.png)

Will not you reproduce over there?
